### PR TITLE
chore: uv to replace poetry

### PR DESCRIPTION
**Pull request by phact**
[`uv`](https://github.com/astral-sh/uv) is a relatively new python package manager with great performance. Langflow has a lot of dependencies that result in slow `poetry` install / lock times which uv can alleviate.

This WIP PR gets the core features like build and unit tests working.

Would be great to get nightlies working with `uv`.

package publish with `uv` is pending.

The current config supports both `poetry` and `uv`. We can remove `poetry` as we gain confidence with `uv`.

### CODE REVIEW

File: scripts/update_dependencies.py
Comment (ogabrielluiz) [inline scripts/update_dependencies.py:1]: I think we can remove this file. It existed because of a missing feature in poetry.

File: Makefile [hunk 1]
```diff
@@ -187,12 +197,12 @@ fix_codespell: ## run codespell to fix spelling errors
...
 format: ## run code formatters
-	@poetry run ruff check . --fix
-	@poetry run ruff format .
+	@uv run ruff check . --fix
+	@uv run ruff format .
 	@cd src/frontend && npm run format
 
 lint: install_backend ## run linters
-	@poetry run mypy --namespace-packages -p "langflow"
+	@uvn run mypy --namespace-packages -p "langflow"
```
Comment (ogabrielluiz) [inline Makefile:205]:
```suggestion
	@uvx run mypy --namespace-packages -p "langflow"
```

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]

**Comment by phact** [comment]
I'm having a look at `make lint` which isn't working

**Review by ogabrielluiz — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -21,6 +21,120 @@ packages = [{ include = "langflow", from = "src/backend" }]
...
+packages = ["src/backend/langflow"]
+
+[project]
+name = "langflow"
+version = "1.0.18"
+description = "A Python package with a built-in web application"
+requires-python = ">=3.10,<3.13"
+# Define your main dependencies here
+dependencies = [
+    "langflow-base==0.0.96",  # Local editable package
```
Comment (jordanrfrazier) [inline pyproject.toml:37]: How does this work if I run `python -m langflow run` in my repo - does it use the pinned 0.0.96 version from pypi or the path from the `tool.uv.sources` above?

File: scripts/update_dependencies.py
Comment (jordanrfrazier) [inline scripts/update_dependencies.py:1]: This file is still needed to update dependencies for the poetry version during docker builds - https://github.com/langflow-ai/langflow/blob/main/docker/build_and_push.Dockerfile#L54

Can we bring this back? Was it causing issues?

File: .github/workflows/python_test.yml [hunk 3]
```diff
@@ -46,23 +46,33 @@ jobs:
...
+      - name: Install the project
+        run: uv sync --dev
       - name: Run unit tests
         uses: nick-fields/retry@v3
         with:
           timeout_minutes: 12
           max_attempts: 2
           command: make unit_tests async=false args="--splits ${{ matrix.splitCount }} --group ${{ matrix.group }}"
+      - name: Minimize uv cache
+        run: uv cache prune --ci
```
Comment (jordanrfrazier) [inline .github/workflows/python_test.yml:75]: I see the `install uv -> python -> restore -> install project .. -> minimize` steps repeated a few times. Can we move that to a sub workflow?

File: scripts/update_dependencies.py
Comment (jordanrfrazier) [reply scripts/update_dependencies.py:1]: (https://github.com/langflow-ai/langflow/actions/runs/11040780042/job/30669541583)

File: pyproject.toml (hunk 2 above)
Comment (phact) [reply pyproject.toml:37]: Good eyes, I *think* we want:
    "langflow-base @ file://./src/backend/base",  # Local editable package

which is what I had originally in here but I don't know if we have a good way of keeping it from getting accidentally committed. uv seems to replace it with the 0.0.96 base automatically when we build.

File: scripts/update_dependencies.py
Comment (phact) [reply scripts/update_dependencies.py:1]: @ogabrielluiz is the plan to update docker too?

File: .github/workflows/python_test.yml (hunk 3 above)
Comment (phact) [reply .github/workflows/python_test.yml:75]: good idea want to take a stab at it?

File: scripts/update_dependencies.py
Comment (phact) [reply scripts/update_dependencies.py:1]: https://github.com/langflow-ai/langflow/pull/3916

File: .github/workflows/python_test.yml (hunk 3 above)
Comment (phact) [reply .github/workflows/python_test.yml:75]: not sure if this is addressed here https://github.com/langflow-ai/langflow/pull/3919

Comment (ogabrielluiz) [reply .github/workflows/python_test.yml:75]: No

File: pyproject.toml (hunk 2 above)
Comment (ogabrielluiz) [reply pyproject.toml:37]: This was fixed. It has to be the version but then add it as a workspace package

File: .github/workflows/python_test.yml (hunk 3 above)
Comment (jordanrfrazier) [reply .github/workflows/python_test.yml:75]: I'll add this once we merge 3919

**Pull request merged**